caffeine/core.py

```diff
@@ -275,22 +318,39 @@ def __toggle_activated(self, note):
         else:
             self.__inhibition_manually_requested = True
 
-        self._performTogglingActions()
-
-        self.status_string == "Caffeine is preventing powersaving."
+        # decide, if we allow the screen to sleep
+        inhibit_screen = False if (self.music_procs > 0 or not self.__inhibition_manually_requested) else True
+        self._performTogglingActions(self.__inhibition_manually_requested, inhibit_screen)
+        logger.info("\n\n")
+        self.status_string = "Caffeine is preventing powersaving."
 
         self.emit("activation-toggled", self.get_activated(),
                   self.status_string)
         self.status_string = ""
 
-    def _performTogglingActions(self):
+    def _performTogglingActions(self, inhibit_suspend, inhibit_screen):
         """This method performs the actions that affect the screensaver and
         powersaving."""
 
         for inhibitor in self.__inhibitors:
             if inhibitor.applicable:
-                logger.info("%s is applicable, running it.", inhibitor)
-                Thread(target=inhibitor.toggle).start()
+                logger.info("%s is applicable, setting it to %s." %(inhibitor, inhibit_suspend))
```

Comment:
This whole set of changes in this `if` are a lot simpler if just using a `inhibits_suspension` attribute on each inhibitor.

Reply:
I will try that out. Thank you.

Reply:
Ok, I propose following solution. I provided the Inhibitor base class with a `is_screen_inhibitor ` property, which gets overridden to return `True` for the inhibitors I had in the second list earlier:
```python
    def is_screen_inhibitor(self):
        return True
```

I also renamed the toggle method of the Inhibitor base class to `set`, which takes a bool. This makes it easier to control them individually, since the Inhibitors now can have a different state one to an other, and toggle would only invert this state:
```python
def set(self, state):
        if state:
            self.inhibit()
        else:
            self.uninhibit()
```

This allows to merge the lists back together and condenses the `_performTogglingActions` method to this:
```python
    def _performTogglingActions(self, suspend, susp_screen):
        for inhibitor in self.__inhibitors:
            if inhibitor.applicable:
                inhibitor.set(susp_screen) if inhibitor.is_screen_inhibitor() \
                                              else inhibitor.set(suspend)
                logger.info("%s is applicable, state: %s"
                            %(inhibitor, inhibitor.running))
        self.__inhibition_successful = not self.__inhibition_successful
```

Reply:
I've updated the pull request with these changes. Please tell me what you think

Reply:
And would it be ok for you to resolve the last couple of travis errors yourself? I don't really get how it is supposed to look and I don't want to spam commits just to solve this.